controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -145,6 +147,67 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
 
+		consumerConfigMap := &corev1.ConfigMap{}
+		consumerConfigMap.Name = r.storageConsumer.Name
+		if r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name != "" {
+			consumerConfigMap.Name = r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name
+		}
+		consumerConfigMap.Namespace = r.namespace
+
+		if err := r.get(consumerConfigMap); client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, err
+		} else if consumerConfigMap.UID == "" {
+			cephCluster, err := util.GetCephClusterInNamespace(r.ctx, r.Client, r.namespace)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if cephCluster.Status.CephStatus == nil || cephCluster.Status.CephStatus.FSID == "" {
+				return reconcile.Result{}, fmt.Errorf("waiting for Ceph FSID")
+			}
+			clientProfileName := util.CalculateMD5Hash([2]string{cephCluster.Status.CephStatus.FSID, r.storageConsumer.Name})
+			if len(clientProfileName) > 36 {
+				// truncating at 36 as that is the maximum limit that csi can use in it's handle
+				clientProfileName = clientProfileName[:36]
+			}
```

Comment:
You are losing the hash guarantees if you truncate. Cant we control the length of the hash as part of the hash generation?

Reply:
CalculateMD5Hash is used at multiple places I don't think we should modify that function maybe create a new one to generate the clientProfile name?

Reply:
I think it's 64 characters